docs/README-develop-best-practices.md

```diff
@@ -43,132 +44,145 @@ This document is intended to assist in developing cisco_node_utils API's that ar
 
-For example, if the default value on all platforms except the n9k is `md5` then set the entry in `command_reference_common.yaml` to `md5` and set the entry in `command_reference_n9k.yaml` to it's default `sha2`.
+```yaml
+message_digest_alg_type:
+  config_get: 'show running interface all'
+  config_get_token: ['/^interface %s$/i', '/^\s*ip ospf message-digest-key \d+ (\S+)/']
+  /N9K/:
+    default_value: 'sha2'
+  else:
+    default_value: 'md5'
+```
+
+See [README_YAML](../lib/cisco_node_utils/cmd_ref/README_YAML.md) for more details about this advanced feature.
 
-### <a name="yaml6">Y6: When possible, use the same `config_get` show command for all properties and document any anomalies.
+### <a name="yaml7">Y7: When possible, use the same `config_get` show command for all properties and document any anomalies.
 
 All properties below use the `show run tacacs all` command except `directed_request` which is documented.
 
-```
-tacacs_server:
-  deadtime:
-    config_get: "show run tacacs all"
-    config_get_token: '/^tacacs-server deadtime\s+(\d+)/'
-    config_set: "%s tacacs-server deadtime %d"
-    default_value: 0
-
-  directed_request:
-    # oddly, directed request must be retrieved from aaa output
-    config_get: "show running aaa all"
-    config_get_token: '/(?:no)?\s*tacacs-server directed-request/'
-    config_set: "%s tacacs-server directed-request"
-    default_value: false
-
-  encryption_type:
-    config_get: "show run tacacs all"
-    config_get_token: '/^tacacs-server key (\d+)\s+(\S+)/'
-    default_value: 0
-
-  encryption_password:
-    config_get: "show run tacacs all"
-    config_get_token: '/^tacacs-server key (\d+)\s+(\S+)/'
-    default_value: ""
+```yaml
+# tacacs_server.yaml
+deadtime:
+  config_get: "show run tacacs all"
+  config_get_token: '/^tacacs-server deadtime\s+(\d+)/'
+  config_set: "%s tacacs-server deadtime %d"
+  default_value: 0
+
+directed_request:
+  # oddly, directed request must be retrieved from aaa output
+  config_get: "show running aaa all"
+  config_get_token: '/(?:no)?\s*tacacs-server directed-request/'
+  config_set: "%s tacacs-server directed-request"
+  default_value: false
+
+encryption_type:
+  config_get: "show run tacacs all"
+  config_get_token: '/^tacacs-server key (\d+)\s+(\S+)/'
+  default_value: 0
+
+encryption_password:
+  config_get: "show run tacacs all"
+  config_get_token: '/^tacacs-server key (\d+)\s+(\S+)/'
+  default_value: ""
 ```
 
-### <a name="yaml7">Y7: Use Key-value wildcards instead of Printf-style wildcards.
+### <a name="yaml8">Y8: Use Key-value wildcards instead of Printf-style wildcards.
 
 The following approach is moderately more complex to implement but is more readable in the ruby code and is flexible enough to handle significant platform differences in CLI. It is therefore the recommended approach for new development.
```

Comment:
Consider:
"Key-value wildcards are moderately more complex to implement than Printf-style wildcards, but they are more readable in the ruby code and are flexible enough to handle significant platform differences in CLI. Key-value wildcards are therefore the recommended approach for new development."